### PR TITLE
Classic UNet model

### PR DESCRIPTION
**Pull request by Nozdi**

### CODE REVIEW

File: src/fast_radiology/models/unet.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+""" Parts of the U-Net model """
```
Comment (grechut) [inline src/fast_radiology/models/unet.py:1]: Maybe add attribution to the original inspiration

File: src/fast_radiology/models/unet.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        return self.double_conv(x)
+
+
+class Down(nn.Module):
+    """Downscaling with maxpool then double conv"""
+
+    def __init__(self, in_channels, out_channels, with_dropout=False):
+        super().__init__()
+        self.maxpool_conv = nn.Sequential(
+            nn.MaxPool2d(2),
```
Comment (grechut) [inline src/fast_radiology/models/unet.py:39]: Initially it looked suspicious, but after review against the original image, putting MaxPool before convolutions here seems to be convenient code solution. In the original image skip-connections happen before MaxPool, which is what will happen here.

File: src/fast_radiology/models/unet.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+        nn.init.kaiming_normal_(self.up.weight)
+        self.conv = DoubleConv(in_channels, out_channels)
+
+    def forward(self, x1, x2):
+        x1 = self.up(x1)
+        # input is CHW
+        diffY = x2.size()[2] - x1.size()[2]
+        diffX = x2.size()[3] - x1.size()[3]
+
+        x1 = F.pad(x1, [diffX // 2, diffX - diffX // 2, diffY // 2, diffY - diffY // 2])
```
Comment (grechut) [inline src/fast_radiology/models/unet.py:64]: I was whether it is needed in our case or after we added padding, and not using `bilinear=True`, maybe we already have a proper size here..
But even if, that line would be obsolete instead of problematic.
I didn't reach that far in review in order to check size of tensor on that level

**Review by grechut — APPROVED** [review]
Generally 👍 

Left 3 minor (only 1.5 actionable) comments
